### PR TITLE
cleanup(pubsub): delete topics older than 3 days

### PR DESCRIPTION
**Pull request by alevenberg**
Ran once, went from 180 topics -> 40. 

I think we might be able to delete some more (cloud-cpp) too, but I will check with someone before adding that

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13556)

### CODE REVIEW

**Review by scotthart — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13556)* status: 0 of 1 files reviewed, 4 unresolved discussions (waiting on @alevenberg)

___
*[`google/cloud/pubsub/samples/topic_admin_samples.cc` line 39 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13556#-Nq-PSHM0TS_zb6toVEc:-Nq-PSHM0TS_zb6toVEd:bm5j3cx) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/3d8841b1c94e069fd227fd255ac618e6b8d6836b/google/cloud/pubsub/samples/topic_admin_samples.cc#L39)):*
> ```cpp
>     google::cloud::pubsub_admin::TopicAdminClient& topic_admin_client,
>     std::string const& project_id, absl::Time time_now) {
>   std::cout << "Cleaning up old topics...\n";
> ```

If this was a debug comment, please remove it. If instead it's adding to the output of the sample code, feel free to leave it.
___
*[`google/cloud/pubsub/samples/topic_admin_samples.cc` line 40 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13556#-Nq-Pq5W066tpr8DL6ri:-Nq-Pq5W066tpr8DL6rj:bp0sbq1) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/3d8841b1c94e069fd227fd255ac618e6b8d6836b/google/cloud/pubsub/samples/topic_admin_samples.cc#L40)):*
> ```cpp
>     std::string const& project_id, absl::Time time_now) {
>   std::cout << "Cleaning up old topics...\n";
>   auto const parent = google::cloud::Project(project_id).FullName();
> ```

Do we need to create a variable here or can the function call be the argument to `ListTopics`?
There's actually a few places where you create a variable and only use it once. They might not all be necessary.
___
*[`google/cloud/pubsub/samples/topic_admin_samples.cc` line 41 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13556#-Nq-PipZ5r6nl-AeMZBh:-Nq-Pip_BvmWoCUgTR45:btzm95v) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/3d8841b1c94e069fd227fd255ac618e6b8d6836b/google/cloud/pubsub/samples/topic_admin_samples.cc#L41)):*
> ```cpp
>   std::cout << "Cleaning up old topics...\n";
>   auto const parent = google::cloud::Project(project_id).FullName();
>   for (auto& topic : topic_admin_client.ListTopics(parent)) {
> ```

Can this be `auto const&`?
___
*[`google/cloud/pubsub/samples/topic_admin_samples.cc` line 44 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13556#-Nq-Rwy-EnXoTp8rFTGk:-Nq-Rwy08c_SnL9Sv47_:bm89o7m) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/3d8841b1c94e069fd227fd255ac618e6b8d6836b/google/cloud/pubsub/samples/topic_admin_samples.cc#L44)):*
> ```cpp
>     if (!topic) continue;
>     auto topic_name = topic->name();
>     std::string keyword = "cloud-cpp-samples";
> ```

This could be `const`.

File: google/cloud/pubsub/samples/topic_admin_samples.cc [hunk 1]
```diff
@@ -28,6 +30,36 @@ using TopicAdminCommand =
...
+    // Extract the date from the resource name which is in the format
+    // `*-cloud-cpp-samples-YYYY-MM-DD-`.
+    auto date =
+        topic_name.substr(topic_name.find(keyword) + keyword.size() + 1, 10);
+
+    absl::TimeZone nyc;
+    if (!absl::LoadTimeZone("America/New_York", &nyc)) continue;
+    absl::CivilDay day;
+    if (absl::ParseCivilTime(date, &day) &&
+        absl::FromCivil(day, nyc) < time_now - absl::Hours(36)) {
```
Comment (devbww) [inline google/cloud/pubsub/samples/topic_admin_samples.cc:55]: Consider using `internal::FormatUtcDate(t)`, just like `RandomTopicId()` does.  Mixing timezones seems bad.

(See `google/cloud/spanner/testing/cleanup_stale_databases.cc` for something similar.)

Comment (alevenberg) [reply google/cloud/pubsub/samples/topic_admin_samples.cc]: Oops, didn't realize. Changed to using UTC timezone

**Review by alevenberg — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13556)* status: 0 of 1 files reviewed, 5 unresolved discussions (waiting on @devbww and @scotthart)

___
*[`google/cloud/pubsub/samples/topic_admin_samples.cc` line 39 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13556#-Nq-PSHM0TS_zb6toVEc:-Nq436Jf-Kr5K0b7Myav:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/3d8841b1c94e069fd227fd255ac618e6b8d6836b/google/cloud/pubsub/samples/topic_admin_samples.cc#L39)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

If this was a debug comment, please remove it. If instead it's adding to the output of the sample code, feel free to leave it.
</blockquote></details>

Done.
___
*[`google/cloud/pubsub/samples/topic_admin_samples.cc` line 40 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13556#-Nq-Pq5W066tpr8DL6ri:-Nq436ic5KVCNn1Eh8Ct:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/3d8841b1c94e069fd227fd255ac618e6b8d6836b/google/cloud/pubsub/samples/topic_admin_samples.cc#L40)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

Do we need to create a variable here or can the function call be the argument to `ListTopics`?
There's actually a few places where you create a variable and only use it once. They might not all be necessary.
</blockquote></details>

Done.
___
*[`google/cloud/pubsub/samples/topic_admin_samples.cc` line 41 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13556#-Nq-PipZ5r6nl-AeMZBh:-Nq4377f1aKI0gkTPhW8:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/3d8841b1c94e069fd227fd255ac618e6b8d6836b/google/cloud/pubsub/samples/topic_admin_samples.cc#L41)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

Can this be `auto const&`?
</blockquote></details>

Done.
___
*[`google/cloud/pubsub/samples/topic_admin_samples.cc` line 44 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13556#-Nq-Rwy-EnXoTp8rFTGk:-Nq437op99JiRxFi8rwB:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/3d8841b1c94e069fd227fd255ac618e6b8d6836b/google/cloud/pubsub/samples/topic_admin_samples.cc#L44)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

This could be `const`.
</blockquote></details>

Done.

**Pull request merged**